### DISCUSSION TITLE
SQLite db file stored in a cloud storage

### DISCUSSION THREAD

**Discussion by cetorres** [fastapi/sqlmodel#625] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class HeroBase(SQLModel):
    name: str = Field(index=True)
    secret_name: str
    age: Optional[int] = Field(default=None, index=True)


class Hero(HeroBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)


class HeroCreate(HeroBase):
    pass


class HeroRead(HeroBase):
    id: int


sqlite_file_name = "database.db"
sqlite_url = f"sqlite:///{sqlite_file_name}"

connect_args = {"check_same_thread": False}
engine = create_engine(sqlite_url, echo=True, connect_args=connect_args)
```

### Description

Is it possible to have the sqlite file stored in a cloud storage, like an AWS or OCI bucket, and read/write the database file from/to there automatically?

### Operating System

Linux, macOS

### SQLModel Version

0.0.8

### Python Version

Python 3.11.1

### Additional Context

I'm creating an API using FastAPI and currently using TinyDB with JSON file stored in an OCI bucket. I'd like to have the same thing with SQLModel and SQLite database file.

**Comment by PookieBuns** [comment, marked as answer]
I don’t think this is a question regarding sqlmodel itself but more about how SQLite works. Generally, if you search around how to access SQLite through a remote server, most people will tell you just don’t. It’s not worth the hassle. SQLite is only accessible through a file and is not meant to be accessed remotely. Use a database that can communicate over TCP like MySQL or Postgres. However, if you really do insist on doing so, you could do some weird hacks like sshfs to the machine storing the db file and communicating the file there. Another really jank way is just to copy the file to your local machine, access it, then copy it back. Anyways, my suggestion would to be just migrate your data to another database that supports remote access
